script.js

```diff
@@ -1,6 +1,243 @@
-function runProgram(){
+let firstQuote;
+let quoteText;
+let authorText;
+let charecterText;
+let missTypes = 0;
+let timer;
+
+async function getUsersName(){
+    let userName = document.querySelector(`#userName`);
+    
+    let name = null;
+    if(localStorage.getItem(`name`) !== null) {
+        name = localStorage.getItem(`name`);
+    }
+    else{
+        name = prompt(`What is your name?`);
+    }
+    userName.textContent = name;
+    localStorage.setItem(`name`, name);
+}
+
+async function getQuote(){
+    const api_url ="https://zenquotes.io/api/quotes/";
+    let data = null
+    try{
+        const response = await fetch(api_url);
+        data = await response.json();
+        console.log(data);
+        displayQuote(data);
+    }
+    catch(error){
+        alert("there has been an error");
+        console.log(error)
+    }
+    
+}
+
+function displayQuote(data){
+    quoteText = document.querySelector('#quote');
+    authorText = document.querySelector('#author');
+    charecterText = document.querySelector('#charecters');
+
+    firstQuote = data[0];
+    console.log(firstQuote)
+    if(firstQuote.c < 116){
+        quoteText.textContent = firstQuote.q;
+        authorText.textContent = firstQuote.a;
+        charecterText.textContent = firstQuote.c;
+    }
+    else{
+        getQuote()
+    }
+
+    document.addEventListener('keyup', typing)
+    let switchColorBtn = document.querySelector('#switchTextColor');
+    switchColorBtn.addEventListener('click', switchTextColor);
+    let reloadBtn = document.querySelector('#reload');
+    reloadBtn.addEventListener('click', reloadSite);
+}
+
+function reloadSite(){
+    location.reload();
+}
+
+function typing(event){
+    const typing = document.querySelector('#typing');
+    missTypes = 0;
+    typingCorrect = true
+    
+    for (let i = 0; i < typing.textContent.length; i++) {
+        if(typing.textContent[i] === firstQuote.q[i]){
+            console.log("You typed this correctly");
+        }
+        else if(typing.textContent[i] !== firstQuote.q[i]){
+            typingCorrect = false
+            missTypes += 1;
+            console.log("You made a miss type! # of misstypes: " + missTypes);
+        }
+    }
+    if (typingCorrect){
+        typing.style.background = 'transparent'
+    }
+    else{
+        typing.style.background = 'red';
+    }
+
+    // START A TIMER:
+    if(typing.textContent.length === 1){
+        console.log("Time started")
+        timer = Date.now();
+    }
+
+    if(typing.textContent === firstQuote.q){
+        // END A TIMER:
+        const elapsedTime = Date.now() - timer;
+        console.log('Elapsed Time:', elapsedTime + 'ms');
+        wordsPerMinuteCalc(elapsedTime);
+    }
+}
```

Comment:
I know that you worked really hard with this, Hale. Your dedication to learning and bettering yourself really shows. While the end result was not exactly what you wanted, I think you can be proud of your dedication and to figuring out a solution that almost got you all the way there/ or was functionally very similar. Great job!